social-api/src/main/java/org/geekbeacon/social/support/JsonHttpHelper.java

```diff
@@ -0,0 +1,37 @@
+package org.geekbeacon.social.support;
+
+import com.jayway.jsonpath.JsonPath;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpResponseException;
+
+/**
+ * Collect of utilities that bring additional functionality beyond the standard library
+ * <p>
+ * ie. wrapping in try/catch and ensuring a default and other misc tooling
+ */
+public final class JsonHttpHelper {
+
+    public static <T> T safeExtract(Object doc, String path, T defaultValue) {
+        try {
+            return JsonPath.read(doc, path);
+        } catch (Exception e) {
+            return defaultValue;
+        }
+    }
+
+    /**
+     * validates that request executed successfully
+     */
+    public static boolean validateRequest(HttpResponse response, boolean useException) throws HttpResponseException {
```

Comment:
Minor: Ambiguous method name.
The method is not checking a request and only the status code of a response, the name should reflect that.
Use `checkResponseCode` or `checkResponseStatusCode` instead.

Reply:
Fixed